common/persistence/visibility/elasticsearch/query/converter.go

```diff
@@ -0,0 +1,406 @@
+		}
+	case "!=":
+		if missingCheck {
+			query = elastic.NewExistsQuery(colName)
+		} else {
+			// Not elastic.NewTermQuery to support String custom search attributes.
+			query = elastic.NewBoolQuery().MustNot(elastic.NewMatchPhraseQuery(colName, colValues[0]))
+		}
+	case "in":
+		query = elastic.NewTermsQuery(colName, colValues...)
+	case "not in":
+		query = elastic.NewBoolQuery().MustNot(elastic.NewTermsQuery(colName, colValues...))
+	case "like":
+		query = elastic.NewMatchPhraseQuery(colName, colValues[0])
+	case "not like":
+		query = elastic.NewBoolQuery().MustNot(elastic.NewMatchPhraseQuery(colName, colValues[0]))
+	}
+
+	return query, nil
+}
+
+func (c *Converter) convertComparisonExprValue(expr sqlparser.Expr) (interface{}, bool, error) {
+	switch e := expr.(type) {
+	case *sqlparser.SQLVal:
+		sqlValue, err := c.parseSqlValue(sqlparser.String(e))
+		return sqlValue, false, err
+	case *sqlparser.GroupConcatExpr:
+		return "", false, fmt.Errorf("%w: 'group_concat'", NotSupportedErr)
+	case *sqlparser.FuncExpr:
+		return "", false, fmt.Errorf("%w: nested func", NotSupportedErr)
+	case *sqlparser.ColName:
+		if sqlparser.String(expr) == "missing" {
+			return "", true, nil
+		}
+		return "", false, fmt.Errorf("%w: column name on the right side of comparison expression", NotSupportedErr)
+	case sqlparser.ValTuple:
+		return sqlparser.String(expr), false, nil
+	default:
+		// cannot reach here
+		return "", false, fmt.Errorf("%w: unexpected value type %T", InvalidExpressionErr, expr)
+	}
+}
+
+// parseSqlRange parses strings like "('1', '2', '3')" which comes from SQL parser.
+func (c *Converter) parseSqlRange(sqlRange string) ([]interface{}, error) {
+	sqlRange = strings.Trim(sqlRange, "()")
+	var values []interface{}
+	for _, v := range strings.Split(sqlRange, ", ") {
+		parsedValue, err := c.parseSqlValue(v)
+		if err != nil {
+			return nil, err
+		}
+		values = append(values, parsedValue)
+	}
+	return values, nil
+}
+
+func (c *Converter) parseSqlValue(sqlValue string) (interface{}, error) {
+	if sqlValue == "" {
+		return "", nil
+	}
+
+	if sqlValue[0] == '\'' && sqlValue[len(sqlValue)-1] == '\'' {
+		strValue := strings.Trim(sqlValue, "'")
+		boolValue, err := strconv.ParseBool(strValue)
+		if err == nil {
+			return boolValue, nil
+		}
+		return strValue, nil
+	}
+
+	// Unquoted value must be a number. Numbers in JSON are float64.
+	floatValue, err := strconv.ParseFloat(sqlValue, 64)
+	if err != nil {
+		return nil, fmt.Errorf("%w: unable to parse %s to float64: %v", InvalidExpressionErr, sqlValue, err)
+	}
+
+	return floatValue, nil
+}
+
+func (c *Converter) convertColName(colNameExpr sqlparser.Expr) (string, error) {
+	colName, isColName := colNameExpr.(*sqlparser.ColName)
+	if !isColName {
+		return "", fmt.Errorf("%w: must be a column name but was %T", InvalidExpressionErr, colNameExpr)
+	}
+
+	colNameStr := sqlparser.String(colName)
+	colNameStr = strings.Replace(colNameStr, "`", "", -1)
```

Comment:
instead of passing -1, why not use ReplaceAll()?

Reply:
And do you need to handle single / double quote?

Reply:
Column name can't have them. And ` is what `sqlparser` uses to escape SQL keywords in column names.